src/Npgsql/BackendMessages/RowDescriptionMessage.cs

```diff
@@ -350,34 +350,39 @@ internal FieldDescription Clone()
         return field;
     }
 
-    internal void GetInfo(Type? type, ref PgConverterInfo lastConverterInfo)
+    internal void GetInfo(Type? type, ref PgConverterInfo lastConverterInfo, out bool asObject)
```

Comment:
When is asObject going to be different than lastConvertInfo.IsBoxingConverter?

Reply:
When the input type to GetFieldValue is typeof(object) we pass the ObjectOrDefaultInfo with asObject marked as true. Otherwise we might try to unbox a `PgConverter<int>` to `PgConverter<object>` while we should just be calling PgConverter.ReadAsObject.